packages/clay-navigation-bar/src/ClayNavigationBar.tsx

```diff
@@ -0,0 +1,197 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import React, {useEffect} from 'react';
+
+import ClayButton from '@clayui/button';
+import ClayIcon from '@clayui/icon';
+import ClayLink from '@clayui/link';
+
+interface Props extends React.HTMLAttributes<HTMLDivElement> {
+	/**
+	 * This property is used to pass a MouseEvent from a NavItem to a
+	 * function when this element is clicked.
+	 */
+	onItemClicked?: (
+		event: React.MouseEvent<HTMLLIElement, MouseEvent>
+	) => void | undefined;
+
+	/**
+	 * The id that will be given to the Navigation<nav> element
+	 */
+	id?: string;
+
+	/**
+	 * An array of items that will be rendered on the dropdown.
+	 */
+	items: Array<{active?: boolean; href?: string; label: string}>;
+
+	/**
+	 * Determines the style of the Navigation Bar
+	 */
+	inverted?: boolean;
+
+	/**
+	 * Path to the location of the spritemap resource.
+	 */
+	spritemap: string;
+}
+
+const ClayNavigationBar: React.FunctionComponent<Props> = ({
+	className,
+	id,
+	inverted = false,
+	items,
+	onItemClicked,
+	spritemap,
+	...otherProps
+}) => {
+	const [visible, setVisible] = React.useState(false);
+	const [transitioning, setTransitioning] = React.useState(false);
```

Comment:
Is it possible to extract some of this transitioning logic and height logic to its own hook? There is a lot of functions being created below here that are a bit hard to follow.

Reply:
Yes! I'm taking a look!

Reply:
I'm taking a look on that :)